### PR TITLE
Add JDBC Vector Store

### PR DESCRIPTION
**Pull request by milderhc**
### Description

Adding JDBC VectorStore and VectorStoreRecordCollection.

MySQL implementation also added.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: api-test/integration-tests/src/test/java/com/microsoft/semantickernel/tests/connectors/memory/jdbc/JDBCVectorStoreRecordCollectionTest.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@Testcontainers
+public class JDBCVectorStoreRecordCollectionTest {
+    @Container
+    private static final MySQLContainer<?> CONTAINER = new MySQLContainer<>("mysql:5.7.34");
```
Comment (dsgrieve) [inline api-test/integration-tests/src/test/java/com/microsoft/semantickernel/tests/connectors/memory/jdbc/JDBCVectorStoreRecordCollectionTest.java:31]: Maybe use [maven resource filtering](https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html) to automagically supply the mysql version.

File: semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/jdbc/JDBCVectorStoreRecordCollection.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+            .then();
+    }
+
+    /**
+     * Creates the collection in the store if it does not exist.
+     *
+     * @return A Mono representing the completion of the creation operation.
+     */
+    @Override
+    public Mono<Void> createCollectionIfNotExistsAsync() {
```
Comment (dsgrieve) [inline semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/jdbc/JDBCVectorStoreRecordCollection.java:114]: Should these CRUD methods return some sort of status? Or should the user expect some sort of SKException if the operation fails? The javadoc should state if SKException is thrown unchecked.

File: semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/recorddefinition/VectorStoreRecordDefinition.java [hunk 3]
```diff
@@ -50,6 +49,52 @@ public List<VectorStoreRecordField> getAllFields() {
...
+            throw new IllegalArgumentException(
+                "Key field not found in record class: " + keyField.getName());
+        }
+    }
+
+    public List<Field> getDataDeclaredFields(Class<?> recordClass) {
+        return getDeclaredFields(
+            recordClass,
+            dataFields.stream().map(f -> (VectorStoreRecordField) f).collect(Collectors.toList()),
+            "Data");
```
Comment (dsgrieve) [inline semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/recorddefinition/VectorStoreRecordDefinition.java:88]: Are "Data" and "Vector" the only two field types allowed? If the field types are limited to a known set, then and enum might be a better choice here.

**Review by dsgrieve — APPROVED** [review]
Excellent body of work, @milderhc!

File: api-test/integration-tests/src/test/java/com/microsoft/semantickernel/tests/connectors/memory/jdbc/JDBCVectorStoreRecordCollectionTest.java (hunk 1 above)
Comment (milderhc) [reply api-test/integration-tests/src/test/java/com/microsoft/semantickernel/tests/connectors/memory/jdbc/JDBCVectorStoreRecordCollectionTest.java:31]: Setting it to latest

**Pull request merged**